main/autogen/autoloader.cc

```diff
@@ -222,7 +222,8 @@ string DefTree::renderAutoloadSrc(const core::GlobalState &gs, const AutoloaderC
             }
             fmt::format_to(std::back_inserter(buf), "}})\n", fullName);
         }
-    } else if (type == Definition::Type::Casgn || type == Definition::Type::Alias) {
+    } else if (type == Definition::Type::Casgn || type == Definition::Type::Alias ||
+               type == Definition::Type::TypeAlias) {
```

Comment:
In the autoloader tree generation process, `TypeAlias` is processed similarly to `Casgn`/`Alias`.